src/math-translator.js

```diff
@@ -284,14 +294,64 @@ function normalizeTranslatedMath(math, lang) {
     return math;
 }
 
-const MathTranslator = {};
+/**
+ * Maybe translate certain math notation to match the notation
+ * that the translator used in the template string.
+ *
+ * This is needed for cases where two or more translations are valid,
+ * e.g. \cdot versus \times), or where there's not a unique mapping from
+ * US notation (intervals and cartesian coordinates have the same notation
+ * in the US but different in many countries).
+ *
+ * The assumption here is that different math notations will not mingle
+ * in a single string, even though in principle a single string can very well
+ * contain both an interval (0,10) and cartesian point (0, 0) for example.
+ * Presumably, such cases are rare.
+ *
+ * Example: String '\\times' will be translated to '\\cdot' if and only if:
+ * 1. lang is in MAYBE_TIMES_AS_CDOT_LOCALES
+ * 2. the 'template' contains '\\cdot' at least once
+ * 3. the 'template' does NOT contain '\\times'
+ *
+ * @param {string} math A math expression to be translated
+ * @param {string} template User-translated template
+ * @param {string} lang The KA locale of the translation language.
+ * @returns {string} translated math expression.
+ */
+function maybeTranslateMath(math, template, lang) {
+    if (!template) {
+        return math;
+    }
 
-MathTranslator.translateMath = translateMath;
-MathTranslator.normalizeTranslatedMath = normalizeTranslatedMath;
-MathTranslator.MATH_RULES_LOCALES = MATH_RULES_LOCALES;
+    const maybeMathTranslations = [
+        // multiplication sign as a centered dot
+        {langs: MATH_RULES_LOCALES.MAYBE_TIMES_AS_CDOT,
+            fromRegex: /\\times/g, toRegex: /\\cdot/, replace: '\\cdot'},
+         // multiplication sign as x
+        {langs: MATH_RULES_LOCALES.MAYBE_CDOT_AS_TIMES,
+            fromRegex: /\\cdot/g, toRegex: /\\times/, replace: '\\times'},
+         // division sign as a colon
+        {langs: MATH_RULES_LOCALES.MAYBE_DIV_AS_COLON,
+            fromRegex: /\\div/g, toRegex: /\\mathbin\{:\}/,
+            replace: '\\mathbin{:}'},
+    ];
+
+    maybeMathTranslations.forEach(function(el) {
+        if (el.langs.includes(lang) &&
+            template.match(el.toRegex) &&
+            !template.match(el.fromRegex)) {
+
+            math = math.replace(el.fromRegex, el.replace);
+
```

Comment:
nit: blank link

Reply:
Hm, I actually like the extra blank line here to separate the if statement from the code, the indentation would not be super clear here.